apps/meteor/server/lib/dataExport/uploadZipFile.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { expect } from 'chai';
+import proxyquire from 'proxyquire';
+import sinon from 'sinon';
+
+// Create stubs for dependencies
+const stubs = {
+	findOneUserById: sinon.stub(),
+	randomId: sinon.stub(),
+	stat: sinon.stub(),
+	getStore: sinon.stub(),
+	insertFileStub: sinon.stub(),
+	createReadStream: sinon.stub(),
+};
+
+const { uploadZipFile } = proxyquire.noCallThru().load('./uploadZipFile.ts', {
+	'@rocket.chat/models': {
+		Users: {
+			findOneById: stubs.findOneUserById,
+		},
+	},
+	'@rocket.chat/random': {
+		Random: {
+			id: stubs.randomId,
+		},
+	},
+	'fs/promises': {
+		stat: stubs.stat,
+	},
+	'fs': {
+		createReadStream: stubs.createReadStream,
+	},
+	'../../../app/file-upload/server': {
+		FileUpload: {
+			getStore: stubs.getStore,
+		},
+	},
+});
+
+describe('Export - uploadZipFile', () => {
+	const randomId = 'random-id';
+	const fileStat = 100;
+	const userName = 'John Doe';
+	const userId = 'user-id';
+	const filePath = 'random-path';
+
+	beforeAll(() => {
+		stubs.findOneUserById.returns({ name: userName });
```

Comment:
Can we add different test cases for each of the options? Since the `userDisplayName` is directly related to what this function returns and it's also directly related to the filename creation, we should also add tests for when the user doesn't have `name` and `username`.

`userDisplayName = user.name || user.username || userId;`

Reply:
Done!